internal/strace/strace.go

```diff
@@ -177,22 +177,20 @@ func (r *Result) recordCommand(cmd, env []string) {
 func (r *Result) parseEnterSyscall(syscall, args string) error {
 	switch syscall {
 	case "write":
-		match := writePattern.FindStringSubmatch(args)
-		// Example: django.template.base\nImporting django.template"..., 0xe64
-		writeBufAndBytesWritten := match[2]
-		for i := len(writeBufAndBytesWritten) - 1; i >= 0; i-- {
-			// TODO: A future PR will use the index right before the comma to get the end of the write buffer.
-			if writeBufAndBytesWritten[i] == ',' {
-				// Use the index after the space and "0x" to get to the first index of the bytes written.
-				bytesWritten, err := strconv.ParseInt(writeBufAndBytesWritten[i+4:len(writeBufAndBytesWritten)], 16, 64)
-				r.recordFileWrite(match[1], bytesWritten)
-				if err != nil {
-					return err
-				}
-				break
-			}
+		// The index of the start of bytes written. Bytes written is expected to be in hex.
+		bytesWrittenHexIndex := strings.LastIndex(args, "0x")
+		// Return an error if we can't find the beginning of bytes written as a hex value or there is no value after the hex prefix.
+		if bytesWrittenHexIndex == -1 || len(args) <= bytesWrittenHexIndex+hexPrefixLength {
+			return fmt.Errorf("strace of file write syscall has the bytes written argument in an unexpected format")
 		}
-
+		// Get the hex value after "0x" to convert to an integer.
+		bytesWritten, err := strconv.ParseInt(args[bytesWrittenHexIndex+hexPrefixLength:len(args)], 16, 64)
+		if err != nil {
+			log.Debug("Could not parse")
```

Comment:
similar to other comment - can just return the error without logging since it's logged at the call site (line 336)